### PR TITLE
Bump Go version to 1.8.3

### PR DESCRIPTION
**Pull request by wojtek-t**
This PR also removed this patched version of Go 1.8.1 which we used to use to workaround performance problem of Go 1.8.1.

Fix https://github.com/kubernetes/kubernetes/issues/45216
Ref #46391

@timothysc @bradfitz

### CODE REVIEW

**Comment by wojtek-t** [comment]
@ixdy - I'm getting this from bazel:
```
W0525 11:20:22.607] ERROR: /workspace/k8s.io/kubernetes/WORKSPACE:35:1: Traceback (most recent call last):
W0525 11:20:22.633] 	File "/workspace/k8s.io/kubernetes/WORKSPACE", line 35
W0525 11:20:22.634] 		go_repositories(go_version = "1.8.3")
W0525 11:20:22.635] 	File "/root/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/external/io_bazel_rules_go/go/private/go_repositories.bzl", line 216, in go_repositories
W0525 11:20:22.636] 		fail("go_repositories: unsupported ve...())))
W0525 11:20:22.674] go_repositories: unsupported version 1.8.3; supported versions are 1.7.5 1.8 1.8.1.
```

Do you know how to fix it?

**Comment by cblecker** [comment]
`build/cni/Makefile` and `cluster/images/etcd/Makefile` actually look like they were missed in the bump to 1.8. Should we bump those too (they're on 1.7.5 right now on master) ?

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -33,7 +33,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_repositories")
 load("@io_bazel_rules_docker//docker:docker.bzl", "docker_repositories")
 
 go_repositories(
-    go_version = "1.8.1",
+    go_version = "1.8.3",
```
Comment (ixdy) [inline build/root/WORKSPACE]: need to update the rules_go dependency to https://github.com/bazelbuild/rules_go/archive/a9df110cf04e167b33f10473c7e904d780d921e6.tar.gz for this, along with updating the sha256.

Comment (ixdy) [inline build/root/WORKSPACE]: Specifically,
64294fd0e74d2aafa03ec3a1f2f9c167e27d17c9a5cf393e8bf79e43258de73d  a9df110cf04e167b33f10473c7e904d780d921e6.tar.gz


So change L1-6 to 
```

http_archive(
    name = "io_bazel_rules_go",
    sha256 = "64294fd0e74d2aafa03ec3a1f2f9c167e27d17c9a5cf393e8bf79e43258de73d",
    strip_prefix = "rules_go-a9df110cf04e167b33f10473c7e904d780d921e6",
    urls = ["https://github.com/bazelbuild/rules_go/archive/a9df110cf04e167b33f10473c7e904d780d921e6.tar.gz"],
)
```

(explanation: the rules_go repo added 1.8.3 in that commit - https://github.com/bazelbuild/rules_go/commit/a9df110cf04e167b33f10473c7e904d780d921e6)

Comment (wojtek-t) [inline build/root/WORKSPACE]: Done.

**Comment by wojtek-t** [comment]
@ixdy - PTAL

**Comment by wojtek-t** [comment]
@k8s-bot pull-kubernetes-node-e2e  test this
@k8s-bot pull-kubernetes-unit  test this

**Review by luxas — APPROVED** [review]
LGTM

**Comment by luxas** [comment]
> build/cni/Makefile and cluster/images/etcd/Makefile actually look like they were missed in the bump to 1.8. Should we bump those too (they're on 1.7.5 right now on master) ?

@cblecker etcd shouldn't use v1.8 yet. `build/cni` now actually is outdated as we're gonna start using the official upstream bins. However, I'll work on that next week.

@wojtek-t I think we could use a release note for this both given the possible crypto issue and the removal of the patch. Agree?

**Comment by cblecker** [comment]
I don't think we've cut a release with go1.8 yet, so would the release note from #41636 suffice?

Thanks for the clarification on etcd/cni, @luxas!

**Comment by wojtek-t** [comment]
@k8s-bot unit test this
@k8s-bot node e2e test this

**Comment by luxas** [comment]
@cblecker Both https://github.com/kubernetes/kubernetes/releases/tag/v1.7.0-alpha.3 and https://github.com/kubernetes/kubernetes/releases/tag/v1.7.0-alpha.4 have been cut since.
I think this deserves a release note. We should also notify the mailing list thread about this.

**Comment by ixdy** [comment]
lgtm2

**Comment by ixdy** [comment]
@k8s-bot pull-kubernetes-node-e2e test this

**Comment by cblecker** [comment]
@k8s-bot pull-kubernetes-node-e2e test this

**Pull request merged**